### DISCUSSION TITLE
Getting events from other devices

### DISCUSSION THREAD

**Discussion by Fabrizz** [kokarare1212/librespot-python#263] [Q&A]
Hi, first of all thanks for the library!

Im currently using this module to get a token (the token generated by librespot acts like a client token so I can get lyrics/canvas/other data). So I was wondering instead of using the normal Spotify Web Api (polling), is there a way to get a event stream from the connection? If i recall correctly it should be a websocket, so updates could be instant.

Also, would this realtime events include data from other devices? 

Thanks in advance!

**Comment by kokarare1212** [comment]
Is a event stream means Spotify Connect?
In those cases, [here](https://github.com/kokarare1212/librespot-python/blob/main/librespot_player/__init__.py) is what is partially implemented.
These are communicated via TCP, not WebSocket.
Specifically, you can prepare a class that implements MessageListener and use session.dealer().add_message_listener to specify the hm url you want to receive to get various events.
